lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -556,9 +615,28 @@ LargeHeapBlock::Mark(void* objectAddress, MarkContext * markContext)
         return;
     }
 
+    if (header->isPageHeapAlloc)
+    {
+        this->VerifyPageHeapPattern();
+    }
+
     DUMP_OBJECT_REFERENCE(markContext->GetRecycler(), objectAddress);
 
-    if (!UpdateAttributesOfMarkedObjects(markContext, objectAddress, header->objectSize, header->GetAttributes(this->heapInfo->recycler->Cookie),
+    unsigned char attributes = header->GetAttributes(this->heapInfo->recycler->Cookie);
+    size_t objectSize = header->objectSize;
+    if (this->InPageHeapMode())
+    {
+        // trim off the trailing part which is not a pointer
+        objectSize = objectSize - (objectSize%sizeof(void*));
```

Comment:
done, and refactored to a function. that always true since size_t is unsigned and objectSize is always positive, right?

---
In reply to: [60994659](https://github.com/Microsoft/ChakraCore/pull/711#discussion_r60994659) [](ancestors = 60994659)